lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -294,12 +311,18 @@ void LowerLayersPass::runOnModuleBody(FModuleOp moduleOp) {
 
       ports.push_back({builder.getStringAttr("_" + operandName.first), refType,
                        Direction::Out, /*sym=*/{}, /*loc=*/loc});
-      body->addArgument(refType, loc);
+      Value replacement = body->addArgument(refType, loc);
       if (isa<RefType>(dest.getType())) {
-        dest.replaceUsesWithIf(body->getArgument(portNum),
-                               [&](OpOperand &operand) {
-                                 return operand.getOwner()->getBlock() == body;
-                               });
+        dest.replaceUsesWithIf(replacement, [&](OpOperand &use) {
+          auto user = use.getOwner();
+          if (!layerBlock->isAncestor(user))
+            return false;
+          if (auto connectLike = dyn_cast<FConnectLike>(user)) {
+            if (use.getOperandNumber() == 0)
+              return true;
+          }
+          return false;
```

Comment:
Consider factoring this to a common static function or lambda so that the PR can capture what it needs.

Reply:
This logic and it's prior use, makes sense to me!